test/Prop.hs

```diff
@@ -990,6 +996,49 @@ monadBind constr eq t (a, b) = withMaxSuccess maxTestCount $
         let list = a >>= \x -> (+ x) <$> b
         listEquals eq stream list
 
+-------------------------------------------------------------------------------
+-- tests for S.groupsBy
+-------------------------------------------------------------------------------
+
+
+-- if list is empty, returns Nothing
```

Comment:
This does not need a doc. If doc should be written start with haddock markup (`-- |`). Left justify all your doc wrapping the lines at 80 columns. You can use some editor command to do that automatically. Start sentences with capital letters and end with a full stop. This applies to all the docs in this PR.